### DISCUSSION TITLE
Hi, I just wondering, what's the reson for choosing QT

### DISCUSSION THREAD

**Discussion by aaabbbdsdsds** [Serial-Studio/Serial-Studio#48] [Q&A] [answered]
I just wanna find some project for pratice.
Unfortunately, I have never finish a project with Qt >.<

**Comment by alex-spataru** [comment, marked as answer]
Hi, I use Qt for developing these kind of projects for the following reasons:

- It eases the process of making cross-platform apps.
- Qt is developed with C++, which results in "native" code for each platform &rarr; faster performance.
- IMO Qt/QML is a very nice GUI toolkit, it has good documentation & I can rest assured that my apps won't break irreversibly as I use newer (minor) versions of Qt. 
- Qt has a lot of handy "modules", such as charts, serial, networking, etc.
- QML (the language used for building "modern" user interfaces) is similar to Javascript, which allows me to design/code the UI very fast, while the "logic" part of the application is implemented in C++.
- The [signals/slots](https://doc.qt.io/qt-5/signalsandslots.html) mechanism is great.
- In my experience, Qt projects are relatively easy to maintain on the longer run.
- You can find many open-source libraries for Qt projects, and the qmake build mechanism is pretty flexible.

[Here](https://medium.com/@yacine.benaffane/why-we-should-use-the-qt-framework-and-the-qml-f01c4edb13cd) is an article explaining more about Qt. Because of my embedded/electronics background, I am more fluent in C & C++ than in other languages. So developing GUI apps with C++ is far easier for me than using Java or another programming language.

The structure of this project is not very complicated, but it's still relatively large. IMHO, it could really use some good documentation & guides to attract possible contributors. 

If you are interested, you are welcome to try to compile the application on your own & explore its inner workings. Please let me know if you have any doubts 😄
